service/worker/scanner/history/scavenger.go

```diff
@@ -339,6 +340,11 @@ func (s *Scavenger) cleanUpWorkflowPastRetention(
 	ctx context.Context,
 	mutableState *persistencepb.WorkflowMutableState,
 ) error {
+	if mutableState.GetExecutionState().GetState() != enums.WORKFLOW_EXECUTION_STATE_COMPLETED {
```

Comment:
Do we need to account for `WORKFLOW_EXECUTION_STATE_VOID` and `WORKFLOW_EXECUTION_STATE_CORRUPTED` here?

Reply:
This is only to clean up past retention data. Currently no plan to clean up corrupted data or in void state.